File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int n = strs.length;

        int pos = 0;
        for (pos = 0; pos < strs[0].length(); pos++) {
            boolean identical = true;
            for (int j = 1; j < n; j++) {
                if (strs[j - 1].length() <= pos || strs[j].length() <= pos) {
                    identical = false;
                    break;
                }
                if (strs[j - 1].charAt(pos) != strs[j].charAt(pos)) {
                    identical = false;
                    break;
                }
            }

            if (!identical) break;
        }

        return strs[0].substring(0, pos);
    }
}

File: 14-longest-common-prefix/longest-common-prefix.scala
object Solution {
    def longestCommonPrefix(strs: Array[String]): String = {
        var firstStr = strs(0);
        var result = firstStr
        var found = false

        for (i <- 0 until firstStr.length if !found) {
            for (j <- 1 until strs.length if !found) {
                if (i >= strs(j).length) {
                    result = firstStr.substring(0, i);
                    found = true
                }
                else if (firstStr(i) != strs(j)(i)) {
                    result = firstStr.substring(0, i);
                    found = true
                }
            }
        }

        result;
    }
}